### PR TITLE
build: bump sinon to a version that supports esm

### PR DESCRIPTION
**Pull request by DLehenbauer**
Attempted at fixing [AB#7319](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7319) and related.  

Fixes:
```
Writing test results relative to package to nyc/junit-report.xml and nyc/junit-report.json

file:///workspaces/f1/packages/common/core-utils/lib/test/promiseCache.spec.js:6
import { useFakeTimers } from "sinon";
         ^^^^^^^^^^^^^
SyntaxError: Named export 'useFakeTimers' not found. The requested module 'sinon' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'sinon';
const { useFakeTimers } = pkg;
```

This bump does not fix the below error. ESM modules cannot be stubbed. 
```
import { stub } from "sinon";
         ^^^^
SyntaxError: Named export 'stub' not found. The requested module 'sinon' is a CommonJS module, which may not support all module.exports as named exports.
```

### CODE REVIEW

**Comment by sonalideshpandemsft** [comment]
Pushed a minor workaround to resolve the error. Feel free to revert the commit if needed :)

File: azure/packages/test/end-to-end-tests/package.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
 		"mocha-multi-reporters": "^1.5.1",
 		"moment": "^2.21.0",
 		"prettier": "~3.0.3",
-		"sinon": "^7.4.2",
+		"sinon": "^17.0.1",
```
Comment (jason-ha) [inline azure/packages/test/end-to-end-tests/package.json:77]: Just a little bump. :)

**Comment by jason-ha** [comment]
>       '@fluidframework/core-interfaces': 2.0.0-internal.8.0.6

Should we roll back these patch bumps?

---
Refers to: pnpm-lock.yaml:15565 in 8194acb. [](commit_id = 8194acb3c7acecf7a76abac8ff62e7c420d3f769, deletion_comment = False)

**Review by jason-ha — APPROVED** [review]
LGTM

**Review by sonalideshpandemsft — APPROVED** [review]

**Pull request merged**